File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        
        int minIndex = 0;
        
        for(int i=0;i<arr.length;i++) {
            if(arr[i] < arr[minIndex]) {
                minIndex = i;
            }
        }
        
        List<Integer> arrList = new ArrayList<>();
        for(int i=0;i<arr.length;i++) {
            if(i!=minIndex) {
                arrList.add(arr[i]);
            }
        }
        
        if(arrList.size()==0) {
            int[] zero = {-1};
            return zero;
        }
        
        int[] answer = new int[arrList.size()];
        for(int i=0;i<arrList.size();i++) {
            answer[i]=arrList.get(i);
        }
        
        return answer;
    }
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.py
import sys

def solution(arr):
    answer = []
    minNum = sys.maxsize
    
    if(len(arr)==1):
        return [-1]
    
    for n in arr:
        if(n<minNum):
            minNum = n
    
    arr.remove(minNum)
    
    return arr